prickly-pufferfish/exec/exec/judge/judge.py

```diff
@@ -0,0 +1,76 @@
+import json
+import pydoc
+from typing import Iterable
+from submission import Submission
+from solution import Solution
+from errors import ImplementationError
+
+
+class Judge:
+    """
+    Class for determining if a solution is correct.
+    It first reads function inputs from a json file, then
+    runs those inputs in both the Solution and Submission
+    class and checks for differences in output.
+    """
```

Comment:
Some good docstrings and comments in this file.  Always happy to see that!  Also great use of type hints, that's always helpful.